### PR TITLE
Allow merge flag to be passed to git checkout

### LINKED ISSUE #3321: git town switch: -m flag
`git town switch` runs `git checkout <branch>`. This fails when there are uncommitted changes that conflict with the new branch. Running [git checkout -m <branch>](https://git-scm.com/docs/git-checkout#Documentation/git-checkout.txt---merge) performs a 3-way merge while switching, thereby allowing to switch branches in more situations.

Add a similar CLI option to `git town switch`. Running `git town switch -m` runs `git checkout -m <branch>`.

### PR DESCRIPTION
**Pull request by sascha-andres**
resolves #3321

### CODE REVIEW

File: src/cmd/switch.go [hunk 1]
```diff
@@ -14,25 +15,42 @@ import (
...
+		if err != nil {
+			panic(fmt.Sprintf("command %q does not have a bool %q flag", cmd.Name(), mergeMessageLong))
+		}
+		return value
+	}
 	cmd := cobra.Command{
 		Use:     "switch",
 		GroupID: "basic",
-		Args:    cobra.NoArgs,
+		Args:    cobra.MaximumNArgs(2),
```
Comment (kevgo) [inline src/cmd/switch.go:41]: I think this setting is for positional arguments only. Can you please try reverting to `NoArgs`? The flag you added should still work.

Background: [git town compress](https://github.com/git-town/git-town/blob/main/src/cmd/compress.go#L45) uses plenty of CLI flags as well and also has `NoArgs`.

File: src/git/frontend_commands.go [hunk 2]
```diff
@@ -37,12 +37,14 @@ func (self *FrontendCommands) AbortRebase() error {
 
 // CheckoutBranch checks out the Git branch with the given name in this repo.
 func (self *FrontendCommands) CheckoutBranch(name gitdomain.LocalBranchName) error {
```
Comment (kevgo) [inline src/git/frontend_commands.go:39]: In my (humble) opinion, you could just add the `merge bool` argument to the existing `CheckoutBranch` method and not create the two additional methods here. Other public methods in this file also have such optional arguments, for example https://github.com/git-town/git-town/blob/main/src/git/frontend_commands.go#L231.

**Review by kevgo — APPROVED** [review]
Thanks so much for this @sascha-andres! 🙏 

This PR implements https://github.com/git-town/git-town/issues/3321, so you could add `resolves #3321` in the body of the PR to automatically close that issue when we ship this PR.

The implementation is great! Below are a few suggestions to move some things to where we store related info in this codebase. If you don't have the bandwidth to implement the comments, just let me know and I'll merge this in and fix it up myself.

I'm excited to use the -m flag for `git town switch`! 🚀

File: src/cmd/switch.go (hunk 1 above)
Comment (sascha-andres) [reply src/cmd/switch.go:41]: NoArgs is defined like this:

`// NoArgs returns an error if any args are included.
func NoArgs(cmd *Command, args []string) error {
	if len(args) > 0 {
		return fmt.Errorf("unknown command %q for %q", args[0], cmd.CommandPath())
	}
	return nil
}
`

I believe ArbitraryArgs is what is required here

File: src/git/frontend_commands.go (hunk 2 above)
Comment (sascha-andres) [reply src/git/frontend_commands.go]: done

File: src/cmd/switch.go [hunk 3]
```diff
@@ -14,25 +15,42 @@ import (
...
+	addMergeFlag := func(cmd *cobra.Command) {
+		cmd.PersistentFlags().BoolP(mergeMessageLong, mergeMessageShort, false, mergeDesc)
+	}
+	readMergeFlag := func(cmd *cobra.Command) bool {
+		value, err := cmd.Flags().GetBool("merge")
+		if err != nil {
+			panic(fmt.Sprintf("command %q does not have a bool %q flag", cmd.Name(), mergeMessageLong))
+		}
+		return value
+	}
```
[the remark this answers is not in the document]
Comment (sascha-andres) [reply src/cmd/switch.go]: done

**Comment by kevgo** [comment]
Thanks again, very well done! 🙏

**Pull request merged**